server/etcdserver/v3_server.go

```diff
@@ -203,18 +204,21 @@ func isTxnSerializable(r *pb.TxnRequest) bool {
 	return true
 }
 
-func isTxnReadonly(r *pb.TxnRequest) bool {
+func checkTxnProperties(r *pb.TxnRequest) (bool, bool) {
```

Comment:
Nit: It's hard to deduct from the name what exactly this function is returning:

type txnProperties {
  //readOnly ...
  readOnly bool   
  //simpleOp is ...
  simpleOp bool   
}